### PR TITLE
migrate Reporta component to ts

### PR DESCRIPTION
**Pull request by juanigaray**
This fixes the following:
- submit function was being re-declared on every render
- formEle could be null, as all elements obtained from document.querySelector
- titulo field param "maxLength" was being passed a string instead of a number

### CODE REVIEW

File: src/components/Reporta.tsx [hunk 1]
```diff
@@ -27,11 +30,7 @@ export default function ReportaForm() {
           </h4>
         </div>
 
-        <form
-          id="formSheet"
-          className={styles.formSheet}
-          onSubmit={(e) => Submit(e)}
-        >
+        <form id="formSheet" className={styles.formSheet} onSubmit={submit}>
```
Comment (xaiki) [inline src/components/Reporta.tsx:33]: any reason why we move frome the onSubmit={Submit} to onSubmit={(e) => Submit(e)} ?

File: src/components/Reporta.tsx [hunk 2]
```diff
@@ -1,16 +1,19 @@
+import { useCallback } from "react";
 import styles from "./Reporta.module.css";
 
 export default function ReportaForm() {
-  function Submit(e) {
-    const formEle = document.querySelector("#formSheet");
+  const submit: React.FormEventHandler<HTMLFormElement> = useCallback((e) => {
+    const formEle = document.querySelector<HTMLFormElement>("#formSheet");
     e.preventDefault;
```
Comment (xaiki) [inline src/components/Reporta.tsx:7]: these seem like confusing notations,
can't it be:
const formEle : HTMLFormElement = document.querySelector() ?

**Review by xaiki — COMMENTED** [review]
to q?

File: src/components/Reporta.tsx (hunk 1 above)
Comment (juanigaray) [reply src/components/Reporta.tsx:33]: Saves a function declaration, which would defeat the purpose of the useCallback

**Comment by juanigaray** [comment]
> to q?

@xaiki could you rephrase that? I don't understand the question

File: src/components/Reporta.tsx (hunk 1 above)
Comment (xaiki) [reply src/components/Reporta.tsx:33]: isn't (e) => Submit(e) a function ?

**Review by xaiki — APPROVED** [review]
lgtm

File: src/components/Reporta.tsx (hunk 2 above)
Comment (juanigaray) [reply src/components/Reporta.tsx:8]: My explanation was wrong! talked it out in group chat and pushed `const formEle : HTMLFormElement | null`
